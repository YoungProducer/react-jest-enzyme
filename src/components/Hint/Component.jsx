import React, { useEffect } from 'react';
import PropTypes from 'prop-types';

const Hint = ({ title, isHintVisible }) => {
	useEffect(() => {
		console.log(isHintVisible);
	}, [isHintVisible]);

	return <div>{isHintVisible && <h1 className="hint">{title}</h1>}</div>;
};

Hint.propTypes = {
	title: PropTypes.string.isRequired,
	isHintVisible: PropTypes.bool.isRequired,
};

export default Hint;
