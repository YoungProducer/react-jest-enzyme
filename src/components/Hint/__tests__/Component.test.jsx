import React from 'react';

import { shallow, mount } from 'enzyme';
import sinon from 'sinon';

import Hint from '../Component';

describe('<Hint />', () => {
	const defaultProps = {
		title: 'Bob no mercy',
	};

	it('is matches snapshot', () => {
		const tree = shallow(<Hint {...defaultProps} />);

		expect(tree).toMatchSnapshot();
	});

	it('display when mouse is over the title', () => {
		const hint = mount(<Hint {...defaultProps} />);

		hint.find('[data-test="title-hover"]').simulate('mouseenter');

		expect(hint.find('.hint').exists()).toEqual(true);

		hint.find('[data-test="title-hover"]').simulate('mouseleave');

		expect(hint.find('.hint').exists()).toEqual(false);
	});
});
