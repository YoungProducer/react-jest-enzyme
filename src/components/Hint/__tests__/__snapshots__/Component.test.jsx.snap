// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hint /> is matches snapshot 1`] = `
<div>
  <h1
    data-test="title-hover"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    Title
  </h1>
</div>
`;
